bench-tps/src/bench.rs

```diff
@@ -538,25 +545,60 @@ pub fn fund_keys<T: Client>(
                 let (blockhash, _fee_calculator) = get_recent_blockhash(client);
 
                 // re-sign retained to_fund_txes with updated blockhash
+                let mut sign_txs = Measure::start("sign_txs");
                 to_fund_txs.par_iter_mut().for_each(|(k, tx)| {
                     tx.sign(&[*k], blockhash);
                 });
+                sign_txs.stop();
+                debug!("sign {} txs: {}us", to_fund_txs.len(), sign_txs.as_us());
 
+                let mut send_txs = Measure::start("send_txs");
                 to_fund_txs.iter().for_each(|(_, tx)| {
                     client.async_send_transaction(tx.clone()).expect("transfer");
                 });
-
-                // retry anything that seems to have dropped through cracks
-                //  again since these txs are all or nothing, they're fine to
-                //  retry
-                for _ in 0..10 {
-                    to_fund_txs.retain(|(_, tx)| !verify_funding_transfer(client, &tx, amount));
+                send_txs.stop();
+                debug!("send {} txs: {}us", to_fund_txs.len(), send_txs.as_us());
+
+                let mut verify_txs = Measure::start("verify_txs");
+                let mut starting_txs = to_fund_txs.len();
+                let mut verified_txs = 0;
+                let mut failed_verify = 0;
+                // Only loop multiple times for small (quick) transaction batches
+                for _ in 0..(if starting_txs < 1000 { 3 } else { 1 }) {
+                    let mut timer = Instant::now();
+                    to_fund_txs.retain(|(_, tx)| {
+                        if timer.elapsed() >= Duration::from_secs(5) {
+                            if failed_verify > 0 {
+                                debug!("total txs failed verify: {}", failed_verify);
+                            }
+                            println!(
+                                "Verifying transfers... {} remaining",
+                                starting_txs - verified_txs
+                            );
+                            timer = Instant::now();
+                        }
+                        let verified = verify_funding_transfer(client, &tx, amount);
+                        if verified {
+                            verified_txs += 1;
+                        } else {
+                            failed_verify += 1;
+                        }
+                        !verified
+                    });
                     if to_fund_txs.is_empty() {
                         break;
                     }
+                    debug!("Looping verifications");
+                    println!("Verifying transfers... {} remaining", to_fund_txs.len());
```

Comment:
nit: I've started to like using `info!` instead of `println!` since you get a nice timestamp

Reply:
Cool, I'm on board with this. Does it make sense to make that change across the board for bench-tps?

Reply:
imo, ✔️

Reply:
Done